lib/adal.js

```diff
@@ -1793,6 +1803,22 @@ var AuthenticationContext = (function () {
         }
     };
 
+    /**
+     * Listens for logouts on other browsing contexts. Clears the user When a logout is detected.
+     * @ignore
+     */
+    AuthenticationContext.prototype._setupCrossBrowsingContextLogoutCleanup = function () {
+        var clearUserOnExternalBrowsingContextLogout = function (evt) {
+            var isExternalBrowsingContextLogout = evt.key === this.CONSTANTS.STORAGE.IDTOKEN && evt.oldValue !== '' && evt.newValue === '';
+
+            if (isExternalBrowsingContextLogout) {
+                this._clearUser();
+            }
+        };
+
+        window.addEventListener('storage', clearUserOnExternalBrowsingContextLogout.bind(this));
```

Comment:
This works on IE and Safari? MDN doesn't have any info: https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event#Browser_compatibility

Reply:
I suppose not a big deal if not.

Reply:
I tested it out in Safari 13 and IE11, both of which seem to work.